src/Avalonia.Layout/LayoutManager.cs

```diff
@@ -133,7 +133,7 @@ private void ExecuteMeasurePass()
 
         private void ExecuteArrangePass()
         {
-            while (_toArrange.Count > 0 && _toMeasure.Count == 0)
+            while (_toArrange.Count > 0)
```

Comment:
Hmm, this doesn't feel like the correct solution. The idea is that when an arrange pass invalidates the measure of some control, the arrange pass is exited and the measure pass run again by [this loop](https://github.com/AvaloniaUI/Avalonia/blob/432795ab3aa34d26360cdbaee211069b837b4aec/src/Avalonia.Layout/LayoutManager.cs#L85).

Are you sure this is the correct fix?

Reply:
Not sure this is the correct fix, but not sure how to improve it if needed.
But there is a problem with MaxPasses in [this loop](https://github.com/AvaloniaUI/Avalonia/blob/432795ab3aa34d26360cdbaee211069b837b4aec/src/Avalonia.Layout/LayoutManager.cs#L85) as they are final count (in our case 3).

So if we have more than 3 visuals than need remeasure in the arrange phase, we have a problem - layout arrange cycle will not be completed for all other visuals in the list (as illustrated in the unit test).

Reply:
Hmm yeah. I'm going to take a deeper look into this.

Reply:
@grokys  can you take a look at #2297 i think i've solved the issues with layoutmanager.
Changes in LayoutManager from #2297 should solve the problems also for this use case